client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -155,12 +175,39 @@ const CheckoutThankYou = React.createClass( {
 		} );
 
 		let purchases = null,
-			wasJetpackPlanPurchased = false;
+			wasJetpackPlanPurchased = false,
+			wasOnlyDotcomPlanPurchased = false;
 		if ( this.isDataLoaded() && ! this.isGenericReceipt() ) {
 			purchases = getPurchases( this.props );
 			wasJetpackPlanPurchased = purchases.some( isJetpackPlan );
+			wasOnlyDotcomPlanPurchased = purchases.every( isPlan );
+		}
+
+		const userCreatedMoment = moment( this.props.user.date );
+		const isNewUser = userCreatedMoment.isAfter( moment().subtract( 2, 'hours' ) );
+		const isInAbTest = abtest( 'paidNuxStreamlined' ) === 'streamlined';
```

Comment:
It is the only a/b test in that area, but I agree that it was not future-proof at all. I used your suggestion.